src/Umbraco.Core/Models/PublicAccessEntry.cs

```diff
@@ -107,9 +107,9 @@ public void RemoveRule(PublicAccessRule rule)
 
         public void ClearRules()
         {
-            foreach (var rule in _ruleCollection)
+            for (var i = _ruleCollection.Count - 1; i >= 0; i--)
```

Comment:
Instead of doing a reverse for loop, you can just do `.ToArray()` - this will copy the `_rulesCollection` to a new collection and doesn't throw the unhandled exception.